tests/unit/single_table/test_base.py

```diff
@@ -104,3 +109,982 @@ def test_fit(self):
+            float_rtol=0.02,
+            progress_bar='progress_bar',
+            output_file_path='output_file_path',
+        )
+        assert expected_call == instance._sample_batch.call_args_list[0]
+        assert expected_call == instance._sample_batch.call_args_list[1]
+
+    def test__conditionally_sample_rows(self):
+        """Test when sampled rows is bigger than 0."""
+        # Setup
+        transformed_data = pd.DataFrame({
+            COND_IDX: [0, 1, 2],
+            'salary.value': [65., 75., 85.]
+        })
+        data = pd.DataFrame({
+            'name': ['Johana', 'Doe', 'Johana Doe'],
+            'salary': [65., 75., 85.],
+        })
+        instance = Mock()
+        instance._sample_in_batches.return_value = data
+        condition = Mock()
+        transformed_condition = Mock()
+
+        # Run
+        result = BaseSynthesizer._conditionally_sample_rows(
+            instance,
+            transformed_data,
+            condition,
+            transformed_condition,
+        )
+
+        # Assert
+        expected_data = pd.DataFrame({
+            'name': ['Johana', 'Doe', 'Johana Doe'],
+            'salary': [65., 75., 85.],
+            COND_IDX: [0, 1, 2]
+        })
+        pd.testing.assert_frame_equal(expected_data, result)
+        instance._sample_in_batches.assert_called_once_with(
+            num_rows=3,
+            batch_size=3,
+            max_tries_per_batch=None,
+            conditions=condition,
+            transformed_conditions=transformed_condition,
+            float_rtol=0.01,
+            progress_bar=None,
+            output_file_path=None
+        )
+
+    def test__conditionally_sample_rows_no_rows_sampled_error(self):
+        """Test that an error is raised when no rows are sampled."""
+        # Setup
+        transformed_data = pd.DataFrame({
+            COND_IDX: [0, 1, 2],
+            'salary.value': [65., 75., 85.]
+        })
+        instance = Mock()
+        instance._sample_in_batches.return_value = []
+        condition = Mock()
+        transformed_condition = 'fancy_condition'
+
+        # Run and Assert
+        expected_message = re.escape(
+            'Unable to sample any rows for the given conditions '
+            '`fancy_condition`. Try increasing `max_tries_per_batch` (currently: None) '
+            'or increasing `batch_size` (currently: 3). Note that '
+            'increasing these values will also increase the sampling time.'
+        )
+        with pytest.raises(ValueError, match=expected_message):
+            BaseSynthesizer._conditionally_sample_rows(
+                instance,
+                transformed_data,
+                condition,
+                transformed_condition,
+                graceful_reject_sampling=False,
+            )
+
+    def test__conditionally_sample_rows_raises_value_error_model_is_gm(self):
+        """Test when sampled rows is lower or 0 and model is ``GaussianMultivariate``."""
```

Comment:
can we improve this docstring to say testing when there are no sampled rows and the model is GaussianMultivariate. Also, wheat is supposed to happen in this test?